eppoclient/client.go

```diff
@@ -2,24 +2,33 @@ package eppoclient
 
 import (
 	"fmt"
+
+	"github.com/Eppo-exp/golang-sdk/v4/eppoclient/applicationlogger"
 )
 
 type Attributes map[string]interface{}
 
 // Client for eppo.cloud. Instance of this struct will be created on calling InitClient.
 // EppoClient will then immediately start polling experiments data from Eppo.
 type EppoClient struct {
-	configRequestor iConfigRequestor
-	poller          poller
-	logger          IAssignmentLogger
+	configRequestor   iConfigRequestor
+	poller            poller
+	logger            IAssignmentLogger
+	applicationLogger applicationlogger.Logger
 }
 
-func newEppoClient(configRequestor iConfigRequestor, poller *poller, assignmentLogger IAssignmentLogger) *EppoClient {
+func newEppoClient(
+	configRequestor iConfigRequestor,
+	poller *poller,
+	assignmentLogger IAssignmentLogger,
+	applicationLogger applicationlogger.Logger,
```

Comment:
making it a required parameter in `newEppoClient` since this is the top level object. elsewhere it is optional